### ISSUE TITLE
Rotary Axis support using true A axis

### ISSUE THREAD

**Issue by funinthefalls** [LaserWeb/LaserWeb4#25]
@tbfleming , @openhardwarecoza asked me to open this issue here in LW4. From the sounds of things, there is no too too much left to do to implement it, just some trig functions for the A values. I would love to have a working rotary axis solution, there are so many project I want to create that require a working rotary axis. I have already built an inexpensive but well built rotary axis setup, just need the firmware to run it now. I will be pushing the fundraising for your K40, I am sure you would love to be able to use a rotary with yours ;)

**Comment by tbfleming** [comment]
I currently have this: http://sherline.com/product/3700-cnc-cnc-ready-4-rotary-table/ paired with http://sherline.com/product/3750-tilting-angle-table/. Will it fit in a K40?

**Comment by funinthefalls** [comment]
All rotarys have an issue with the clearance under the nozzle. Even using the lightobjects rotary which was designed for the K40 only allows about a 2" diameter object that will fit under the laser nozzle. Solution is to cut out a big hole in the bottom of the case, raise up the machine on extended sides at the bottom.

**Comment by gndprx** [comment]
Will this work for both laser and cnc? I have a rotary on the way that I plan to make portable between the two machines.

**Comment by tbfleming** [comment]
Laser raster should be straight forward. I'll have to think about the mill ops and other laser ops.

**Comment by funinthefalls** [comment]
@tbfleming  I took a look at the links you posted for the rotary, that thing is a monster for a K40. If you want, I can give you some links for some 3d printed or laser cut rotary attachments that are a bet fit size wise.

**Comment by tbfleming** [comment]
@funinthefalls I definitely need recommendations. I created a new thread: https://plus.google.com/101442607030198502072/posts/MFNhLYfoTtC

**Comment by openhardwarecoza** [comment]
Good enough to close for now?

**Comment by tbfleming** [comment]
So far no one's tested it. May malfunction horribly.

**Comment by cojarbi** [comment]
Hope I can get to test this week, having trouble with my board and firmware

**Comment by openhardwarecoza** [comment]
(; maiming oneself could be a new issue? Closed as #wontfix

**Comment by funinthefalls** [comment]
I am waiting for a stepper motor that gets here tomorrow, will test then.

**Comment by tbfleming** [comment]
> (; maiming oneself could be a new issue? Closed as #wontfix

Are you saying we should drop rotary support?

**Comment by openhardwarecoza** [comment]
No,  i am saying if something goes wrong (; its a new issue hehe

On Dec 19, 2016 8:33 PM, "Todd Fleming" <notifications@github.com> wrote:

(; maiming oneself could be a new issue? Closed as #wontfix

Are you saying we should drop rotary support?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/LaserWeb/LaserWeb4/issues/25#issuecomment-268041179>,
or mute the thread

**Comment by tbfleming** [comment]
I don't consider it finished; the approach I used isn't compatible with TinyG. I'm waiting to see how well it works (if at all) with smoothie so I have a baseline to compare against when I move it to a different approach.

**Comment by openhardwarecoza** [comment]
Noted.  Keeping issue open :)

**Comment by gndprx** [comment]
Definitely keep it open. Rotary is a huge part of my go-forward plans for both CNC and Laser. I have my rotary sitting in the shop waiting for me to rewire everything.

Looking forward to having a nice integrated package :)

**Comment by tbfleming** [comment]
No testers; code may have rotted. Might remove it.

**Comment by FabCreator** [comment]
Give us a few weeks. Does it hurt to leave it in?

**Comment by gndprx** [comment]
I can have my rotary up in a few days assuming I can get a few days in the shop where I don't have other more pressing items. I have everything I need and just need to spend some time wiring.

**Comment by gndprx** [comment]
Ok, let me rephrase. I can have my rotary up as soon as I can figure out why I'm not able to drive the stepper.

**Comment by openhardwarecoza** [comment]
There is a specific build time switch to enable rotary (or we can bug
@wolfmanjm to give you a built bin)

**Comment by tbfleming** [comment]
#304 duplicated this, but I didn't catch the duplication in time. Discussion moved there.
